### PR TITLE
feat: improve Relation field type tag

### LINKED ISSUE #2359: Object Detail - Improve Relation field type tag
## Context

See #2340 

For this issue:
- Change the "Relation" field type tag in the Object Detail fields table to display:
  - The plural name of the related object
  - A "1:1" icon for the "Has one" relation types
  - A "1:N" icon for the "Has many/Belongs to one" relation type
- Clicking on the tag navigates to the related object's detail page

## Figma

Image

https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=11338%3A64076&mode=design&t=XfNCHa1nYfOXgsMW-1

### Scenario: Clicking on a "Relation" field type tag

**GIVEN** I am on an Object Detail page
**WHEN** I click on a "Relation" field type tag
**THEN** I navigate to the related object's Object Detail page

### PR DESCRIPTION
**Pull request by thaisguigon**
Closes #2359

### CODE REVIEW

File: front/src/modules/settings/data-model/object-details/components/SettingsObjectFieldDataType.tsx [hunk 1]
```diff
@@ -24,21 +40,33 @@ const StyledDataType = styled.div<{ value: FieldMetadataType }>`
...
   const theme = useTheme();
 
-  const { label, Icon } = dataTypes?.[value];
-
   return (
-    <StyledDataType value={value}>
-      <Icon size={theme.icon.size.sm} />
-      {label}
+    <StyledDataType onClick={onClick} value={value}>
+      <ClassNames>
```
Comment (charlesBochet) [inline front/src/modules/settings/data-model/object-details/components/SettingsObjectFieldDataType.tsx:59]: this is introducing a new pattern, we should discuss it :)
Is there another nice way to achieve the same goal?

File: front/src/modules/settings/data-model/object-details/components/SettingsObjectFieldItemTableRow.tsx [hunk 2]
```diff
@@ -32,27 +36,50 @@ const StyledIconTableCell = styled(TableCell)`
...
   const theme = useTheme();
-  const { Icon } = useLazyLoadIcon(fieldItem.icon ?? '');
+  const { Icon } = useLazyLoadIcon(fieldMetadataItem.icon ?? '');
+  const navigate = useNavigate();
 
   // TODO: parse with zod and merge types with FieldType (create a subset of FieldType for example)
-  const fieldDataTypeIsSupported = Object.keys(dataTypes).includes(
-    fieldItem.type,
-  );
+  const fieldDataTypeIsSupported = fieldMetadataItem.type in dataTypes;
```
Comment (charlesBochet) [inline front/src/modules/settings/data-model/object-details/components/SettingsObjectFieldItemTableRow.tsx:46]: I think dataTypes should be renamed to SettingsFieldMetadataTypes

**Review by charlesBochet — COMMENTED** [review]
Left two comments!

File: front/src/modules/settings/data-model/object-details/components/SettingsObjectFieldItemTableRow.tsx (hunk 2 above)
Comment (thaisguigon) [reply front/src/modules/settings/data-model/object-details/components/SettingsObjectFieldItemTableRow.tsx]: Done

File: front/src/modules/settings/data-model/object-details/components/SettingsObjectFieldDataType.tsx (hunk 1 above)
Comment (thaisguigon) [reply front/src/modules/settings/data-model/object-details/components/SettingsObjectFieldDataType.tsx]: As discussed: I've wrapped the Icon with `styled` instead.

**Review by charlesBochet — APPROVED** [review]
Thank you!

**Pull request merged**